### PR TITLE
[FHB-732] Disable EF Core runtime migrations

### PR DESCRIPTION
**Pull request by Top-Cat**
The migrations were never meant to run in production.
It just turns out our apps can't tell they're in production....

In every environment other than Development the environment is "Release"
So I've removed all the checks for production as they were all doing literally nothing and also removed calls to MigrateAsync.

### CODE REVIEW

File: src/service/service-directory-api/tests/FamilyHubs.ServiceDirectory.Api.FunctionalTests/CustomWebApplicationFactory.cs [hunk 1]
```diff
@@ -53,6 +54,15 @@ public void SetupTestDatabaseAndSeedData()
         {
             var context = scopedServices.GetRequiredService<ApplicationDbContext>();
 
+            // I wish I could just call Migrate() but our migrations don't work on sqlite?
```
Comment (Zac-Digital) [inline src/service/service-directory-api/tests/FamilyHubs.ServiceDirectory.Api.FunctionalTests/CustomWebApplicationFactory.cs:57]: FYI on this https://learn.microsoft.com/en-us/ef/core/providers/sqlite/limitations#migrations-limitations

TL;DR SQLite sucks (well, Microsoft think so)

**Review by StuwiiDev — APPROVED** [review]

File: src/service/service-directory-api/tests/FamilyHubs.ServiceDirectory.Api.FunctionalTests/CustomWebApplicationFactory.cs (hunk 1 above)
Comment (Top-Cat) [reply src/service/service-directory-api/tests/FamilyHubs.ServiceDirectory.Api.FunctionalTests/CustomWebApplicationFactory.cs:57]: It's not just a sqlite problem, we have explicitly set mssql specific column types. Maybe we'd run into some other problems after we sorted that, maybe not....
